dashboard/app/helpers/levels_helper.rb

```diff
@@ -47,6 +47,11 @@ def set_videos_and_blocks_and_callouts_and_instructions
 
     if current_user && params[:load_previous]
       @start_blocks = current_user.last_attempt(@level).try(:level_source).try(:data)
+    elsif current_user && params[:level_source_id]
+      @level_source = LevelSource.find(params[:level_source_id])
```

Comment:
Ah gotcha.  Yeah Blockly isn't currently set up in a way that it can be reloaded easily without a fresh page load.  Can we set `appOptions.level.startBlocks` before calling `initApp()`?